### DISCUSSION TITLE
Importing the flake part seems to define `perSystem.devshells.<name>`

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#179] [Q&A]
Seems that merely importing the nci flake part module allows defining `perSystem.devshells.<name>`.
I'm not sure how severe this is. For one, we were wondering why our shell doesn't work ("flake does not provide attribute devShells"). Of course, our error was neglecting to import the numtide devshell flake part, but it would have been more useful if we had gotten an error about that, specifically.

Seems like there's a comment about the relevant code [here](https://github.com/yusdacra/nix-cargo-integration/blob/c8e10f50dcfa9e58fe52221a4fc986cc7869098e/src/implementation.nix/#L10-L20), but perhaps author was not aware of this particular possible consequence.

:shrug:

**Comment by 90-008** [comment]
how exactly did you get this error message? i'm not aware of anything like that

**Reply by mightyiam** [reply]
If the devshell flake module is not imported and nci flake module is imported one would get this error.

**Reply by 90-008** [reply]
oh do you mean if numtideDevshell is set but devshell flake module is not imported? can probably mention that in the option documentation yes.

for error im not sure how that would work, could perhaps check a if an option numtide flake module adds exists though it sounds like it could cause infinite recursion potentially (and not to mention these options could change on numtide side so i would rather not depend on that). checking inputs if a numtide input exists, maybe, but people may want to change the name of the input or another url or etc. so that's not a good solution either imo. i don't think it would be really necessary either way though if we just make the documentation clear about it

**Reply by mightyiam** [reply]
Can't a `mkIf` solve the issue? I might be way off here.
